DOCSP-34448 - DocumentDB comparison page (#937)
mongodb/docs-ecosystem: source/documentdb-support.txt

.. _documentdb-support:

===============================
Amazon DocumentDB Compatibility
===============================

.. facet::
   :name: genre
   :values: reference
 
.. meta:: 
   :keywords: AWS, support

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1

.. figure:: /figures/documentdb.png
   :alt: Amazon DocumentDB logo

`Amazon DocumentDB (with MongoDB compatibility) <https://docs.aws.amazon.com/documentdb/latest/developerguide/what-is.html>`__
is a fast, reliable, and fully managed database service. Amazon DocumentDB makes it easy
to set up, operate, and scale MongoDB-compatible databases in the cloud.
With Amazon DocumentDB, you can run the same application code and use the same drivers
and tools that you use with MongoDB.

Version Information
-------------------

> Amazon DocumentDB supports MongoDB v4.0 and MongoDB v5.0. These versions lack
> support for the newer features and functionality in the versions of MongoDB available
> on `MongoDB Atlas. <https://www.mongodb.com/docs/atlas/>`__

Reviewer: Suggestion: Since this is a marketing-type doc, I would avoid using words with negative associations like "lack".

```suggestion
Amazon DocumentDB supports MongoDB v4.0 and MongoDB v5.0. These legacy versions do not 
support some of the newer features available on `MongoDB Atlas. <https://www.mongodb.com/docs/atlas/>`__, 
but additional capabilities and management tools are available with Amazon DocumentDB.
```

Note: What I wrote is a guess, but the point is to focus on benefits, not limitations.

Author: Good point.

.. tip:: MongoDB Version History
   
   To learn more about the features available in each version of MongoDB, see 
   `MongoDB Evolved - Version History. <https://www.mongodb.com/evolved>`__

Compatibility
-------------

As of November 2023, Amazon DocumentDB is only about 34 percent compatible with the
MongoDB API.
On Amazon DocumentDB, the following MongoDB features are

> available only in a limited capacity or are not supported at all:

Reviewer: Suggestion: redundant? 
```suggestion
available in a limited capacity or are not supported at all:
```

> - Features: `Capped collections <https://www.mongodb.com/docs/manual/core/capped-collections/>`__,

Reviewer: Suggestion: Bolding the list items might make this more legible.

```suggestion
- *Features*: `Capped collections <https://www.mongodb.com/docs/manual/core/capped-collections/>`__,
```

Author: yep, meant to do that, actually

  ``$toBool``, ``$toDate``, ``$toDecimal``, ``$toDouble``, ``$toInt``, ``$toLong``,
  ``$toObjectId``, ``$toString``, ``$isNumber``, ``$rand``, ``$sampleRate``, ``$getField``
- Aggregation Stages: ``$collStats``, ``$facet``, ``$bucket``, ``$bucketAuto``,
  ``$sortByCount``, ``$listLocalSessions``, ``$listSessions``, ``$graphLookup``

To learn more about Amazon DocumentDB's compatibility with MongoDB v4.0 and MongoDB v5.0,
see the following AWS documentation:

- `MongoDB Compatibility <https://docs.aws.amazon.com/documentdb/latest/developerguide/compatibility.html#mongodb-50>`__
- `Functional Differences: Amazon DocumentDB and MongoDB <https://docs.aws.amazon.com/documentdb/latest/developerguide/functional-differences.html>`__
- `Supported MongoDB APIs, Operations, and Data Types <https://docs.aws.amazon.com/documentdb/latest/developerguide/mongo-apis.html>`__

For the most current compatibility status between Amazon DocumentDB and MongoDB, see
`Is Amazon DocumentDB Compatible with MongoDB Atlas? <https://www.isdocumentdbreallymongodb.com/>`__

Support
-------

MongoDB doesn't offer commercial support for Amazon DocumentDB. For help with this product,

> contact AWS Support directly.

Reviewer: Question: Do we have a link for AWS Support?
